services/algolia-sync/src/components/algolia/message-sync.js

```diff
@@ -0,0 +1,50 @@
+const algoliasearch = require('algoliasearch');
+const gql = require('graphql-tag');
+const apollo = require('../graphql/query');
+const { buildSections, boostResult } = require('./helpers');
+const { ALGOLIA_APPID, ALGOLIA_API_KEY } = require('../../env');
+
+const { log } = console;
+
+const client = algoliasearch(ALGOLIA_APPID, ALGOLIA_API_KEY);
+
+const query = message => (gql`
+  query {
+      content(input:{ id: ${message.id}, status: any }) {
+        id
+        type
+        created
+        published
+        updated
+        status
+        websiteSchedules { section { hierarchy {fullName} }}
+        primaryImage { name filePath fileName src isLogo }
+        primarySite{ host }
+        name
+        teaser
+        body
+      }
+  }
+`);
+
+const upsertToIndex = async (message) => {
+  log(`Proccessing: ${message.id}`);
+  const c = await apollo.queryFromBase(query(message), message.tenant);
+  try {
+    if (c.websiteSchedules) {
+      c.set(await buildSections(c));
+    }
+
+    c.boost = await boostResult(c);
+
+    const index = client.initIndex(message.tenant);
+    index.saveObject({
+      objectID: c.content.id,
+      ...c.content,
+    });
+  } catch (e) {
+    throw (`Failed on: ${message.id}`, e);
```

Comment:
Only throw an Error object. If you're looking to change the error message just do:
```js
throw new Error(`Failed on: ${message.id}`);
```

Reply:
Ok, I fixed it.